app/src/test/java/org/simple/clinic/recentpatientsview/RecentPatientsViewControllerTest.kt

```diff
@@ -139,6 +140,90 @@ class RecentPatientsViewControllerTest {
                 updatedAtRelativeTimestamp = Yesterday
             ),
             gender = FEMALE
+        )
+    ))
+    verify(screen).showOrHideRecentPatients(isVisible = true)
+  }
+
+  @Test
+  fun `when number of recent patients is greater than recent patient limit then add see all item`() {
+    val patientUuid1 = UUID.randomUUID()
+    val patientUuid2 = UUID.randomUUID()
+    val patientUuid3 = UUID.randomUUID()
+    val patientUuid4 = UUID.randomUUID()
+    whenever(patientRepository.recentPatients(facility.uuid, recentPatientLimit)).thenReturn(Observable.just(listOf(
+        PatientMocker.recentPatient(
+            uuid = patientUuid1,
+            fullName = "Ajay Kumar",
+            age = Age(42, Instant.now(), LocalDate.MIN),
+            gender = TRANSGENDER,
+            lastBp = LastBp(systolic = 127, diastolic = 83, recordedAt = Instant.now())
+        ),
+        PatientMocker.recentPatient(
+            uuid = patientUuid2,
+            fullName = "Vijay Kumar",
+            age = Age(24, Instant.now(), LocalDate.MIN),
+            gender = MALE,
+            lastBp = null
+        ),
+        PatientMocker.recentPatient(
+            uuid = patientUuid3,
+            fullName = "Vinaya Kumari",
+            age = Age(27, Instant.now(), LocalDate.MIN),
+            gender = FEMALE,
+            lastBp = LastBp(systolic = 142, diastolic = 72, recordedAt = Instant.now().minus(1, ChronoUnit.DAYS))
+        ),
+        PatientMocker.recentPatient(
+            uuid = patientUuid4,
+            fullName = "Abhilash Devi",
+            age = Age(37, Instant.now(), LocalDate.MIN),
+            gender = TRANSGENDER,
+            lastBp = LastBp(systolic = 139, diastolic = 92, recordedAt = Instant.now().minus(3, ChronoUnit.DAYS))
+        )
+    )))
+
+    uiEvents.onNext(ScreenCreated())
+
+    verify(screen).updateRecentPatients(listOf(
+        RecentPatientItem(
```

Comment:
Can you extract creation of `RecentPatientItem` to variables and use that in both returning from `patientRepository.recentPatients(facility.uuid, recentPatientLimit)` and verifying it here?

Reply:
I realize that this change is more involved and this works for now.
Maybe we can think of doing this in the future.